articles/openshift/oshift-how-build-app-private-repo.md

```diff
@@ -0,0 +1,127 @@
+---
+title: How to build an OpenShift application from a private GitHub repository | UKCloud Ltd
+description: Shows how to build an OpenShift application where the source code is in a private GitHub repository secured with an SSH key pair
+services: openshift
+author: Andy Watkins
+
+toc_rootlink: How To
+toc_sub1:
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Build an OpenShift application from a private GitHub repo
+toc_fullpath: How To/oshift-how-build-app-private-repo.md
+toc_mdlink: oshift-how-build-app-private-repo.md
+---
+
+# How to build an OpenShift application from a private GitHub repo
+
+## Overview
+
+With a private GitHub repo, you want to ensure it stays hidden from prying
+eyes, but you also want your OpenShift application to be able to build from the repository.
+The way around this is to use a 
+[deploy key](https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys).
+
+This guide assumes you have access to the command-line `oc` client, and have
+logged in to your OpenShift instance with `oc login`
+
+
+## Building an OpenShift application from a private GitHub repository
+1. Create an ssh key-pair
+
+    > [!NOTE]
+    > Don't use your regular one, and make sure you don't overwrite your regular one either.
+     
+    ```bash
+    ssh-keygen -t rsa -b 4096 -c "jbloggs@example.com" -f my_GitHub_deploy_key
+    ```
+    
+    This will generate both the private and public key files. The public one will
+    have a `.pub` suffix.
+
+2. Add the pubic key
```

Comment:
Typo. Should be public.
Also, I think my original comment may have been confusing. No need to duplicate the phrase "Add the public key". This should just be:
2. Add the public key to the repository as a Deploy key (instructions with screenshots on GitHub).